### DISCUSSION TITLE
Allow for more property types to be public

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [livewire/livewire#2304] [3. Feature ideas]
My public properties on a component can only be some types:

> Properties can ONLY be either JavaScript-friendly data types (string, int, array, boolean), OR one of the following PHP types: Stringable, Collection, DateTime, Model, EloquentCollection.

But what if I want to allow other resources? Now I have to extend them with `\DateTime` to get it working, but it would be nice if it just was possible.

**Comment by ju5t** [comment]
It's something you have to be specific about as there needs to be some conversion between the two. What specific properties are you looking for?

**Reply by jeroenvanrensen** [reply]
Hi @ju5t,

I would like to have a class that does not extends any other class in my blade view. If I don't use Livewire it works because I can render anything in plain blade. 

But if I do use Livewire I cannot use any random class, but I have to extend it with `DateTime` to get it working. This means it is possible for Livewire to use any PHP class, but it simply does not allow for it.

I hope now you understand. Jeroen

**Reply by ju5t** [reply]
Can you give examples of the classes that you need? What kind of functionality do they have?

**Reply by jeroenvanrensen** [reply]
This is a simplified example:

```php
<?php

namespace X;

class SomeClass
{
    public $text = 'Hi there!';

    public function getText()
    {
        return $this->text;
    }
}
```

**Reply by ju5t** [reply]
Isn't passing them to the view in `render()` sufficient? You change properties of the class as they will change, but if you need to do that you probably need to store them as well. It does allow you to interact with any class you require though.

**Reply by jeroenvanrensen** [reply]
Yeah that's smart! Thanks!

**Comment by boynet** [comment]
Thanks for the tip.. 
I just extended all my class I needed to share with Component to make it work

**Reply by jeroenvanrensen** [reply]
Happy to help 👍

**Reply by francoism90** [reply]
@boynet @jeroenvanrensen Could you please give an example how to extend? I want to add my custom types as well.

**Reply by ju5t** [reply]
Please read https://laravel-livewire.com/docs/2.x/properties#wireable-properties. This is the documented way to use custom types.
